gpii/node_modules/transformer/src/js/Transformer.js

```diff
@@ -293,4 +293,93 @@ fluid.registerNamespace("gpii.transformer");
+     * 'transformSpec.hasElemVal.isTrue', otherwise the value will be 'transformSpec.hasElemVal.isFalse'.
+     *
+     * @param {Array} arr An array with the elements that will be transformed into the tagged array.
+     * @param {Object} transformSpec The required spec for the transform. It should have the form:
+     *  {
+     *     "tag": "tagValue",
+     *     "elems": [
+     *     ],
+     *     "hasElemVal": {
+     *         "isTrue": isTrueValue,
+     *         "isFalse": isFalseValue
+     *     }
+     *   }
+     * @return {Array} A tagged array result of processing the input array with the provided 'transformSpec'.
+     */
+    gpii.transformer.jsonArrayToTaggedArray = function (arr, transformSpec) {
+        var output = [];
+
+        // Check the transformSpec
+        gpii.transformer.jsonArrayToTaggedArray.checkSpec(transformSpec);
+
+        fluid.each(transformSpec.elems, function (elem) {
+            var arrayElem = {};
+
+            var hasElem = arr.indexOf(elem) !== -1;
+            fluid.set(arrayElem, "$t", elem);
+
+            if (hasElem) {
+                fluid.set(arrayElem, transformSpec.tag, transformSpec.hasElemVal.isTrue);
+            } else {
+                fluid.set(arrayElem, transformSpec.tag, transformSpec.hasElemVal.isFalse);
+            }
+
+            output.push(arrayElem);
+        });
+
+        return output;
+    };
+
+    /**
+     * Checks that the provied specification have the proper form, if the provided spec doesn't
+     * have the correct form, 'fluid.fail()' will be called.
+     *
+     * @param {Object} transformSpec The transform spec to be validated.
+     */
+    gpii.transformer.jsonArrayToTaggedArray.checkSpec = function (transformSpec) {
+        if (!transformSpec.tag) {
+            fluid.fail("jsonArrayToTaggedArray requires a 'tags' block set");
+        }
+        if (!transformSpec.elems || !fluid.isArrayable(transformSpec.elems)) {
+            fluid.fail("jsonArrayToTaggedArray requires a 'elems' block array");
+        }
+        if (transformSpec.hasElemVal) {
+            if (!transformSpec.hasElemVal.isTrue) {
+                fluid.fail("jsonArrayToTaggedArray 'hasElemVal' block requires a 'isTrue' value set");
+            }
+            if (!transformSpec.hasElemVal.isFalse) {
+                fluid.fail("jsonArrayToTaggedArray 'hasElemVal' block requires a 'isFalse' value set");
+            }
+        } else {
+            fluid.fail("jsonArrayToTaggedArray requires a 'hasElemVal' set");
+        }
+    };
+
+    /**
+     * Realizes the inverse operation than 'gpii.transformer.jsonArrayToTaggedArray'. Recieves
+     * a tagged array, and returns just the values present in that array.
+     *
+     * @param {Array} arr The tagged array which values are going to be extracted.
+     * @return {Array} An array with the values of the provied tagged array.
+     */
+    gpii.transformer.jsonArrayToTaggedArray.invert = function (arr) {
```

Comment:
Inverter is faulty, recall that the goal of the inverse transform is to invert the configuration, not to run the inverse transform. Please supply a test for this, that is, one which actually invokes fluid.model.transform.invertConfiguration to verify the round-trip, as per https://github.com/fluid-project/infusion/blob/master/tests/framework-tests/core/js/ModelTransformationTests.js#L91

Reply:
Yes, I have now implemented what I think it's a proper invert transform.